src/filesystem/drive/directory_handle.rs

```diff
@@ -32,6 +32,13 @@ pub struct DirectoryHandle {
 }
 
 impl DirectoryHandle {
+
+    pub async fn to_entry(&self) -> Result<DirectoryEntry, OperationError> {
```

Comment:
I'd drop the `to_` prefix on this, generally in rust `to_*` methods consume `self` which isn't what's going on here.